homeassistant/components/zwave_js/update.py

```diff
@@ -0,0 +1,197 @@
+) -> None:
+    """Set up Z-Wave button from config entry."""
+    client: ZwaveClient = hass.data[DOMAIN][config_entry.entry_id][DATA_CLIENT]
+
+    @callback
+    def async_add_firmware_update_entity(node: ZwaveNode) -> None:
+        """Add firmware update entity."""
+        driver = client.driver
+        assert driver is not None  # Driver is ready before platforms are loaded.
+        async_add_entities([ZWaveNodeFirmwareUpdate(driver, node)], True)
+
+    config_entry.async_on_unload(
+        async_dispatcher_connect(
+            hass,
+            f"{DOMAIN}_{config_entry.entry_id}_add_firmware_update_entity",
+            async_add_firmware_update_entity,
+        )
+    )
+
+
+def get_latest_firmware(
+    firmwares: list[FirmwareUpdateInfo],
+) -> FirmwareUpdateInfo:
+    """Get the latest firmware from the available updates."""
+    latest_firmware = FirmwareUpdateInfo("0.0.0", "", [])
+    for firmware in firmwares:
+        if AwesomeVersion(firmware.version) > AwesomeVersion(latest_firmware.version):
+            latest_firmware = firmware
+    return latest_firmware
+
+
+class ZWaveNodeFirmwareUpdate(UpdateEntity):
+    """Representation of a firmware update entity."""
+
+    _attr_entity_category = EntityCategory.CONFIG
+    _attr_device_class = UpdateDeviceClass.FIRMWARE
+    _attr_supported_features = (
+        UpdateEntityFeature.INSTALL
+        | UpdateEntityFeature.RELEASE_NOTES
+        | UpdateEntityFeature.SPECIFIC_VERSION
+    )
+    _attr_has_entity_name = True
+
+    def __init__(self, driver: Driver, node: ZwaveNode) -> None:
+        """Initialize a Z-Wave device firmware update entity."""
+        self.driver = driver
+        self.node = node
+        self.available_firmware_updates: list[FirmwareUpdateInfo] = []
+        self._latest_version_firmware: FirmwareUpdateInfo | None = None
+
+        # Entity class attributes
+        self._attr_name = "Firmware"
+        self._base_unique_id = get_valueless_base_unique_id(driver, node)
+        self._attr_unique_id = f"{self._base_unique_id}.firmware_update"
+        # device may not be precreated in main handler yet
+        self._attr_device_info = DeviceInfo(
+            identifiers={get_device_id(driver, node)},
+            sw_version=node.firmware_version,
+            name=node.name or node.device_config.description or f"Node {node.node_id}",
+            model=node.device_config.label,
+            manufacturer=node.device_config.manufacturer,
+            suggested_area=node.location if node.location else None,
+        )
+
+        self._attr_installed_version = self._attr_latest_version = node.firmware_version
+
+    @property
+    def extra_state_attributes(self) -> Mapping[str, Any] | None:
+        """Return entity specific state attributes."""
+
+        return {
+            ATTR_AVAILABLE_FIRMWARE_UPDATES: [
+                {
+                    "version": firmware.version,
+                    "release_notes": firmware.changelog,
```

Comment:
Are the release notes useful for automations? Are these links?

Reply:
just text in markdown format. I'm not sure if there is a limit on this. Maybe this doesn't need to belong here. I would ask though, where would a user go to learn about the different versions available so they can understand the differences? Maybe we need a firmware DB? @AlCalzone thoughts?

Reply:
No limits so far. Some examples can be found in https://github.com/zwave-js/firmware-updates/blob/9c8e4ef0e5700907487e867961930dd9323b16a6/firmwares/enbrighten-ge/ZW3004_14295.json#L20 and the other files.

I expect most of them to be similar, see e.g. [Zooz](https://www.support.getzooz.com/kb/article/1152-zse18-motion-sensor-change-log/) or [Inovelli](https://support.inovelli.com/portal/en/kb/articles/firmware-change-log-lzw31-dimmer-switch-black-series#V157_-_06292021) for examples.

I think showing the changes where the update is makes sense, so users can easily decide whether they need it.

Reply:
I don't think we should add the release notes to the extra state attributes. We'll show the release notes for the available update.

Reply:
got it - maybe a feature request then to add a URL to link to to get more details? Right now if a user wants to install a version other than the latest, they'd have to chase down what that includes.